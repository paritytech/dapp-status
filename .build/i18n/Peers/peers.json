[
  {
    "id": "peers.acceptNonReserved.label",
    "defaultMessage": "Accept non-reserved"
  },
  {
    "id": "peers.dropNonReserved.label",
    "defaultMessage": "Drop non-reserved"
  },
  {
    "id": "peers.addReserved.label",
    "defaultMessage": "Add reserved"
  },
  {
    "id": "peers.removeReserved.label",
    "defaultMessage": "Remove reserved"
  },
  {
    "id": "status.peers.title",
    "defaultMessage": "network peers"
  },
  {
    "id": "status.peers.table.header.id",
    "defaultMessage": "ID"
  },
  {
    "id": "status.peers.table.header.remoteAddress",
    "defaultMessage": "Remote Address"
  },
  {
    "id": "status.peers.table.header.name",
    "defaultMessage": "Name"
  },
  {
    "id": "status.peers.table.header.ethHeader",
    "defaultMessage": "Header (ETH)"
  },
  {
    "id": "status.peers.table.header.ethDiff",
    "defaultMessage": "Difficulty (ETH)"
  },
  {
    "id": "status.peers.table.header.caps",
    "defaultMessage": "Capabilities"
  },
  {
    "id": "peers.form.label",
    "defaultMessage": "Peer enode URL"
  },
  {
    "id": "peers.form.action.label",
    "defaultMessage": "{add, select, true {Add} false {}}{remove, select, true {Remove} false {}}"
  },
  {
    "id": "peers.form.cancel.label",
    "defaultMessage": "Cancel"
  },
  {
    "id": "peers.acceptNonReservedPeers.success",
    "defaultMessage": "Accepting non-reserved peers"
  },
  {
    "id": "peers.dropNonReservedPeers.success",
    "defaultMessage": "Dropping non-reserved peers"
  },
  {
    "id": "peers.form.action.success",
    "defaultMessage": "Successfully {add, select, true {added} false {}}{remove, select, true {removed} false {}} a reserved peer"
  }
]