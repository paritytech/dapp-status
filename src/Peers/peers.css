/* Copyright 2015-2017 Parity Technologies (UK) Ltd.
/* This file is part of Parity.
/*
/* Parity is free software: you can redistribute it and/or modify
/* it under the terms of the GNU General Public License as published by
/* the Free Software Foundation, either version 3 of the License, or
/* (at your option) any later version.
/*
/* Parity is distributed in the hope that it will be useful,
/* but WITHOUT ANY WARRANTY; without even the implied warranty of
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
/* GNU General Public License for more details.
/*
/* You should have received a copy of the GNU General Public License
/* along with Parity.  If not, see <http://www.gnu.org/licenses/>.
*/

.peers {
  margin-top: 0.5em;
  overflow: auto;
  width: 100%;

  table {
    border-collapse: collapse;
    width: 100%;
  }

  th {
    padding: 0.5em;
    text-align: left;
    white-space: nowrap;
  }
}

.peer {
  &:nth-child(odd) {
    background-color: rgba(200, 200, 200, 0.1);
  }

  td {
    border-top: 1px solid #333;
    font-size: 0.9em;
    overflow: hidden;
    padding: 0.5em 0.25em;
    white-space: nowrap;
  }
}

.form {
  align-items: center;
  display: flex;
  flex-direction: row;
  float: right;
  width: 40em;

  .input {
    flex: 1;
    margin-right: 1em;
  }
}
