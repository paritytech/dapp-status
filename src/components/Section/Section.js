// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import PropTypes from 'prop-types';

import Header from 'semantic-ui-react/dist/commonjs/elements/Header';
import Segment from 'semantic-ui-react/dist/commonjs/elements/Segment';

import styles from './Section.css';

const Section = ({ children, title, ...rest }) => (
  <Segment padded {...rest}>
    <Header as="h3">{title}</Header>
    {children}
  </Segment>
);

export default Section;
