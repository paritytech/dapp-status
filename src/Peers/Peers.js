// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { inject, observer } from 'mobx-react';
import Button from 'semantic-ui-react/dist/commonjs/elements/Button';
import Table from 'semantic-ui-react/dist/commonjs/collections/Table';
import ScrollableText from '@parity/ui/lib/ScrollableText';
import ShortenedHash from '@parity/ui/lib/ShortenedHash';
import { FormattedMessage } from 'react-intl';

import ReservedPeersForm from './ReservedPeersForm';
import Section from '../components/Section';
import styles from './Peers.css';

export class Peers extends Component {
  state = {
    showSettings: false
  };

  static propTypes = {
    netPeersStore: PropTypes.object.isRequired
  };

  handleToggleSettings = () =>
    this.setState({
      showSettings: !this.state.showSettings
    });

  renderRow = (peer, index) => {
    const { caps, id, name, network, protocols } = peer;
    return (
      <Table.Row key={id}>
        <Table.Cell>{index + 1}</Table.Cell>
        <Table.Cell>
          <ScrollableText text={id} />
        </Table.Cell>
        <Table.Cell>{network.remoteAddress}</Table.Cell>
        <Table.Cell>{name}</Table.Cell>
        <Table.Cell>
          <ShortenedHash data={protocols.eth.head} />
        </Table.Cell>
        <Table.Cell>
          {protocols.eth.difficulty.gt(0) &&
            protocols.eth.difficulty.toExponential(16)}
        </Table.Cell>
        <Table.Cell>{caps && caps.join(', ')}</Table.Cell>
      </Table.Row>
    );
  };

  render() {
    const { netPeersStore: { realPeers } } = this.props;

    return (
      <Section
        title={
          <div>
            <FormattedMessage
              id="dapp.status.peers.title"
              defaultMessage="Network Peers"
            />
            <Button
              basic
              circular
              className={styles.settingsButton}
              compact
              icon="setting"
              onClick={this.handleToggleSettings}
              size="large"
            />
          </div>
        }
      >
        {this.state.showSettings && <ReservedPeersForm />}
        <Table celled selectable striped>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell />
              <Table.HeaderCell singleLine>
                <FormattedMessage
                  id="dapp.status.peers.idLabel"
                  defaultMessage="ID"
                />
              </Table.HeaderCell>
              <Table.HeaderCell>
                <FormattedMessage
                  id="dapp.status.peers.remoteAddressLabel"
                  defaultMessage="Remote Address"
                />
              </Table.HeaderCell>
              <Table.HeaderCell>
                <FormattedMessage
                  id="dapp.status.peers.nameLabel"
                  defaultMessage="Name"
                />
              </Table.HeaderCell>
              <Table.HeaderCell>
                <FormattedMessage
                  id="dapp.status.peers.ethHeaderLabel"
                  defaultMessage="Header (ETH)"
                />
              </Table.HeaderCell>
              <Table.HeaderCell>
                <FormattedMessage
                  id="dapp.status.peers.ethDiffLabel"
                  defaultMessage="Difficulty (ETH)"
                />
              </Table.HeaderCell>
              <Table.HeaderCell>
                <FormattedMessage
                  id="dapp.status.peers.capsLabel"
                  defaultMessage="Capabilities"
                />
              </Table.HeaderCell>
            </Table.Row>
          </Table.Header>
          <Table.Body>{realPeers.map(this.renderRow)}</Table.Body>
        </Table>
      </Section>
    );
  }
}

export default inject('netPeersStore')(observer(Peers));
