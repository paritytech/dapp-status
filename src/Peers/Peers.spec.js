// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import BigNumber from 'bignumber.js';

import { Peers } from './Peers';
import Section from '../components/Section';

const props = {
  netPeersStore: {
    netPeers: {
      active: BigNumber(1),
      connected: BigNumber(2),
      max: BigNumber(3)
    },
    realPeers: [
      {
        caps: ['Foo'],
        id: 'Bar',
        name: 'Baz',
        network: { remoteAddress: 'Qux' },
        protocols: { eth: { difficulty: BigNumber(20), head: '0x123' } }
      }
    ]
  }
};

test('should render correctly without settings', () => {
  const component = shallow(<Peers {...props} />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should render correctly when showing settings', () => {
  const component = shallow(<Peers {...props} />);
  component.setState({ showSettings: true });

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should handle toggle settings', () => {
  const component = shallow(<Peers {...props} />);
  const button = component.find(Section).props().title.props.children[1];

  button.props.onClick();
  expect(component.state('showSettings')).toBe(true);

  button.props.onClick();
  expect(component.state('showSettings')).toBe(false);
});
