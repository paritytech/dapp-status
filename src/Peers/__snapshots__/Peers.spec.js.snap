// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when showing settings 1`] = `
<Section
  title={
    <div>
      <FormattedMessage
        defaultMessage="Network Peers"
        id="dapp.status.peers.title"
        values={Object {}}
      />
      <Button
        as="button"
        basic={true}
        circular={true}
        className={undefined}
        compact={true}
        icon="setting"
        onClick={[Function]}
        size="large"
      />
    </div>
  }
>
  <inject-ReservedPeersForm-with-netPeersStore />
  <Table
    as="table"
    celled={true}
    selectable={true}
    striped={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
        />
        <TableHeaderCell
          as="th"
          singleLine={true}
        >
          <FormattedMessage
            defaultMessage="ID"
            id="dapp.status.peers.idLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Remote Address"
            id="dapp.status.peers.remoteAddressLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Name"
            id="dapp.status.peers.nameLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Header (ETH)"
            id="dapp.status.peers.ethHeaderLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Difficulty (ETH)"
            id="dapp.status.peers.ethDiffLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Capabilities"
            id="dapp.status.peers.capsLabel"
            values={Object {}}
          />
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="Bar"
      >
        <TableCell
          as="td"
        >
          1
        </TableCell>
        <TableCell
          as="td"
        >
          <ScrollableText
            text="Bar"
          />
        </TableCell>
        <TableCell
          as="td"
        >
          Qux
        </TableCell>
        <TableCell
          as="td"
        >
          Baz
        </TableCell>
        <TableCell
          as="td"
        >
          <ShortenedHash
            data="0x123"
          />
        </TableCell>
        <TableCell
          as="td"
        >
          2.0000000000000000e+1
        </TableCell>
        <TableCell
          as="td"
        >
          Foo
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Section>
`;

exports[`should render correctly without settings 1`] = `
<Section
  title={
    <div>
      <FormattedMessage
        defaultMessage="Network Peers"
        id="dapp.status.peers.title"
        values={Object {}}
      />
      <Button
        as="button"
        basic={true}
        circular={true}
        className={undefined}
        compact={true}
        icon="setting"
        onClick={[Function]}
        size="large"
      />
    </div>
  }
>
  <Table
    as="table"
    celled={true}
    selectable={true}
    striped={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
        />
        <TableHeaderCell
          as="th"
          singleLine={true}
        >
          <FormattedMessage
            defaultMessage="ID"
            id="dapp.status.peers.idLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Remote Address"
            id="dapp.status.peers.remoteAddressLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Name"
            id="dapp.status.peers.nameLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Header (ETH)"
            id="dapp.status.peers.ethHeaderLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Difficulty (ETH)"
            id="dapp.status.peers.ethDiffLabel"
            values={Object {}}
          />
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          <FormattedMessage
            defaultMessage="Capabilities"
            id="dapp.status.peers.capsLabel"
            values={Object {}}
          />
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="Bar"
      >
        <TableCell
          as="td"
        >
          1
        </TableCell>
        <TableCell
          as="td"
        >
          <ScrollableText
            text="Bar"
          />
        </TableCell>
        <TableCell
          as="td"
        >
          Qux
        </TableCell>
        <TableCell
          as="td"
        >
          Baz
        </TableCell>
        <TableCell
          as="td"
        >
          <ShortenedHash
            data="0x123"
          />
        </TableCell>
        <TableCell
          as="td"
        >
          2.0000000000000000e+1
        </TableCell>
        <TableCell
          as="td"
        >
          Foo
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</Section>
`;
